src/common/containers/build-details.js

```diff
@@ -43,33 +62,44 @@ class BuildDetails extends Component {
 
 }
 BuildDetails.propTypes = {
-  account: PropTypes.string.isRequired,
-  repo: PropTypes.string.isRequired,
-  fullName: PropTypes.string.isRequired,
+  repository: PropTypes.shape({
+    owner: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    fullName: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired
+  }),
   buildId: PropTypes.string.isRequired,
   build: PropTypes.object,
   isFetching: PropTypes.bool,
+  snapLink: PropTypes.string,
   error: PropTypes.object,
   dispatch: PropTypes.func.isRequired
 };
 
 const mapStateToProps = (state, ownProps) => {
-  const account = ownProps.params.account.toLowerCase();
-  const repo = ownProps.params.repo.toLowerCase();
+  const owner = ownProps.params.owner.toLowerCase();
+  const name = ownProps.params.name.toLowerCase();
+  const fullName = `${owner}/${name}`;
+  const url = getGitHubRepoUrl(fullName);
```

Comment:
> Who should trigger this action? Component in componentWillMount ?
Yes. This is how we loaded pre-requisite data like user name, balance, payment methods etc on the last React App I worked on.